From 08d91a9b13839ebbe004c237ee339ca4b0490433 Mon Sep 17 00:00:00 2001
From: Ying Zhang <ying.zhang22455@nxp.com>
Date: Thu, 29 Sep 2016 11:16:16 +0800
Subject: [PATCH 109/115] armv8: aarch32: Add KVM support for AArch32 on ARMv8

This patch is to add KVM support for AArch32 execution state on ARMv8.

Signed-off-by: Alison Wang <alison.wang@nxp.com>
---
 arch/arm/include/asm/cputype.h |    2 ++
 arch/arm/kvm/guest.c           |    1 +
 2 files changed, 3 insertions(+)

diff --git a/arch/arm/include/asm/cputype.h b/arch/arm/include/asm/cputype.h
index 85e374f..bc045b1 100644
--- a/arch/arm/include/asm/cputype.h
+++ b/arch/arm/include/asm/cputype.h
@@ -76,6 +76,8 @@
 #define ARM_CPU_PART_CORTEX_A15		0x4100c0f0
 #define ARM_CPU_PART_MASK		0xff00fff0
 
+#define ARM_CPU_PART_CORTEX_AARCH32_A53 0x4100d030
+
 #define ARM_CPU_XSCALE_ARCH_MASK	0xe000
 #define ARM_CPU_XSCALE_ARCH_V1		0x2000
 #define ARM_CPU_XSCALE_ARCH_V2		0x4000
diff --git a/arch/arm/kvm/guest.c b/arch/arm/kvm/guest.c
index 3705fc2..6d9022b 100644
--- a/arch/arm/kvm/guest.c
+++ b/arch/arm/kvm/guest.c
@@ -247,6 +247,7 @@ int __attribute_const__ kvm_target_cpu(void)
 {
 	switch (read_cpuid_part()) {
 	case ARM_CPU_PART_CORTEX_A7:
+	case ARM_CPU_PART_CORTEX_AARCH32_A53:
 		return KVM_ARM_TARGET_CORTEX_A7;
 	case ARM_CPU_PART_CORTEX_A15:
 		return KVM_ARM_TARGET_CORTEX_A15;
-- 
1.7.9.5

